src/backend/mmap.rs

```diff
@@ -0,0 +1,151 @@
+extern crate memmap;
+extern crate failure;
+
+use self::failure::ResultExt;
+
+use super::Backend;
+
+use ::error;
```

Comment:
`use` statements in Rust are always absolute, so `::error` and `error` are equivalent, please consider removing the colons